src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_validators.py

```diff
@@ -50,20 +52,27 @@ def validate_registry_name_create(namespace):
 def validate_storage_account_name(namespace):
     client = storage_client_factory().storage_accounts
 
-    if namespace.storage_account_name is None:
+    if namespace.storage_account_name:
+        storage_account_name = namespace.storage_account_name
+        if arm_get_storage_account_by_name(storage_account_name) is None:
+            raise CLIError('The storage account {} does not exist'.format(storage_account_name))
```

Comment:
Doesn't RP already validate this? No need to add a ton of CLI side logic for these things usually. (They go out of sync, hard to rev)